### PR TITLE
virtualx.eclass: EAPI-8 support

### PR DESCRIPTION
**Pull request by a17r**
Prerequisite of https://github.com/gentoo/kde/pull/903.

### CODE REVIEW

File: eclass/virtualx.eclass [hunk 1]
```diff
@@ -6,17 +6,16 @@
...
-	6|7) ;;
-	*) die "virtualx.eclass: EAPI ${EAPI} is not supported yet." ;;
+case ${EAPI} in
+	6|7|8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} is not supported." ;;
 esac
 
-if [[ ! ${_VIRTUAL_X} ]]; then
-_VIRTUAL_X=1
+if [[ ! ${_VIRTUAL_X_ECLASS} ]]; then
```
Comment (SoapGentoo) [inline eclass/virtualx.eclass:17]:
```suggestion
if [[ ! ${_VIRTUALX_ECLASS} ]]; then
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Comment by a17r** [comment]
Not sure what to make with the remaining gentoo-dev review for the eclass...

File: eclass/virtualx.eclass [hunk 2]
```diff
@@ -33,17 +32,16 @@ _VIRTUAL_X=1
 # @DESCRIPTION:
 # Dep string available for use outside of eclass, in case a more
 # complicated dep is needed.
-# You can specify the variable BEFORE inherit to add more dependencies.
-VIRTUALX_DEPEND="${VIRTUALX_DEPEND}
+# DEPRECATED: Pre-EAPI-8 you can specify the variable BEFORE inherit
+# to add more dependencies.
+[[ ${EAPI} == [67] ]] && VIRTUALX_DEPEND="${VIRTUALX_DEPEND}"
```
Comment (mgorny) [inline eclass/virtualx.eclass:37]: Err, you do realize that this doesn't do anything, right? ;-)

File: eclass/virtualx.eclass [hunk 3]
```diff
@@ -33,17 +32,16 @@ _VIRTUAL_X=1
...
+VIRTUALX_DEPEND+="
```
Comment (mgorny) [inline eclass/virtualx.eclass:38]: …and appending effectively means it still works as it used to? ;-)

Maybe reverse the condition and set it to empty value instead?

**Review by mgorny — CHANGES_REQUESTED** [review]

File: eclass/virtualx.eclass (hunk 2 above)
Comment (a17r) [reply eclass/virtualx.eclass:37]: rright...

File: eclass/virtualx.eclass [hunk 4]
```diff
@@ -30,20 +29,20 @@ _VIRTUAL_X=1
 : ${VIRTUALX_REQUIRED:=test}
 
 # @ECLASS-VARIABLE: VIRTUALX_DEPEND
+# @DEFAULT_UNSET
```
Comment (mgorny) [inline eclass/virtualx.eclass:32]:
```suggestion
# @OUTPUT_VARIABLE
```

File: eclass/virtualx.eclass [hunk 5]
```diff
@@ -30,20 +29,20 @@ _VIRTUAL_X=1
...
 # @DESCRIPTION:
-# Dep string available for use outside of eclass, in case a more
-# complicated dep is needed.
-# You can specify the variable BEFORE inherit to add more dependencies.
-VIRTUALX_DEPEND="${VIRTUALX_DEPEND}
+# Standard dependencies string that is automatically added to BDEPEND
+# (in EAPI-6: DEPEND) unless VIRTUALX_REQUIRED is set to "manual".
+# DEPRECATED: Pre-EAPI-8 you can specify the variable BEFORE inherit
+# to add more dependencies.
+[[ ${EAPI} == [67] ]] || VIRTUALX_DEPEND=""
```
Comment (mgorny) [inline eclass/virtualx.eclass:38]:
```suggestion
[[ ${EAPI} != [67] ]] && VIRTUALX_DEPEND=""
```

It is generally to easier to read this as 'if ... then ...'.

File: eclass/virtualx.eclass [hunk 6]
```diff
@@ -30,20 +29,20 @@ _VIRTUAL_X=1
...
+VIRTUALX_DEPEND+="
 	x11-base/xorg-server[xvfb]
 	x11-apps/xhost
 "
+[[ ${EAPI} == [67] ]] || readonly VIRTUALX_DEPEND
```
Comment (mgorny) [inline eclass/virtualx.eclass:43]:
```suggestion
[[ ${EAPI} != [67] ]] && readonly VIRTUALX_DEPEND
```

**Review by mattst88 — APPROVED** [review]
Looks good to me. Thanks!

**Pull request closed without merging**